Implement jsonb containment operators

zaneduffield opened the pull request:
Implemented jsonb containment operators `<@` and `@>` as described in #3032.
First time contributing so please forgive any mistakes.

pksunkara commented on diesel/src/pg/expression/expression_methods.rs:
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
+    }
+    /// Creates a PostgreSQL `<@` expression.
+    ///
+    /// This operator returns whether a JSON value is contained by another JSON value. `foo.contains(bar)` is the
```
```suggestion
    /// This operator checks whether left hand side JSONB value is contained by right hand side JSON value. `foo.contains(bar)` is the
```

Please break up the sentence into lines accordingly after the length changes.

zaneduffield replied:
fixed

zaneduffield commented on diesel/src/pg/expression/expression_methods.rs:
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
     }
+    /// Creates a PostgreSQL `@>` expression.
+    ///
+    /// This operator returns whether a JSON value contains another JSON value
```
fixed

zaneduffield commented on diesel/src/pg/expression/expression_methods.rs:
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
+    {
+        Grouped(ContainsJsonb::new(self, other.as_expression()))
+    }
+    /// Creates a PostgreSQL `<@` expression.
```
fixed

zaneduffield commented:
I'm not sure why that "sqlite bundled" test is failing now. The error makes it seem unlreated to my changes.

weiznich commented:
Yes, that's an unrelated CI failure. Likely due to a updated nightly version that introduced a new warning.

pksunkara reviewed (changes requested):
My comments are still not addressed. Are you sure you pushed correctly?

zaneduffield commented:
> My comments are still not addressed. Are you sure you pushed correctly?

It looks like I somehow reverted my changes which addressed them in the merge commit. I'll fix this when I get home today.

zaneduffield commented:
I fixed the issue, but also realised that fixup commits on merge commits require you to rebase and do the merge again. To clean up this mess a bit I've rebased to latest master, resolving merge conflicts, and force pushed to this branch.

pksunkara commented:
@weiznich not sure if you want to merge or not because of that SQLite issue

weiznich commented:
It's fine to merge it, I will try to address the sqlite issue today or tomorrow in a separate PR.
